src/Paket.Core/Common/Domain.fs

```diff
@@ -96,31 +96,26 @@ type QualifiedPackageName =
         let packageName = PackageName packageName
         QualifiedPackageName (groupName, packageName)
 
+type PackageMatch(ex:String) =
+    member this.Expression = Regex("^" + ex + "$", RegexOptions.CultureInvariant ||| RegexOptions.IgnoreCase) 
```

Comment:
Are you sure this no longer creates regex objects on every access?

Reply:
pretty much :) 
as far I understand F# this should be constructor syntax, filling the Expression member with result of the expression after equals
and I'm seeing the described improvement on my machine, getting filtered update in approximately same time as full one

Reply:
Decompiled into C#:

```C#
		[CompilationMapping(SourceConstructFlags.ObjectType)]
		[Serializable]
		public class PackageMatch
		{
			public PackageMatch(string ex) : this()
			{
				this.ex = ex;
			}

			public Regex Expression
			{
				get
				{
					return new Regex("^" + this.ex + "$", RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
				}
			}

			internal string ex;
		}
```

Reply:
I know there is a syntax to initialize it only once, I think it is something with ``val``. Tbh, in that regard F# is really confusing.

Reply:
And would it maybe make sense to add ``RegexOptions.Compiled`? https://stackoverflow.com/questions/513412/how-does-regexoptions-compiled-work#7707369

Reply:
Here's a sharplab.io example: https://sharplab.io/#v2:DYLgZgzgPg9gDgUwHYAIDKBPCAXBBbAWAChjsNEUBhACgQA8QUcAnASyQHMBKFAXmJQo8+AEYJmKAG4BDYCgCidOMwQQIrGKl4p6KANQoARCOnNDAwYNLkEVAPoAlBAGMYeOAFdctBk2xtOHn4iQWE8MQlsAAtWCAA6RWVVdU0+HTp9IxMzIA===

You can see here that thanks to `member val` the property value is computed once on construction and then just reused, whereas for `C_Recompute` the property is recomputed each access.

Reply:
Note that we don't have access to `System.Text.RegularExpressions` in sharplab so that's why I didn't repro your scenario exactly.

Reply:
Oh my... :) Ok, I'll revise it to create the instance only once for sure. 
At this moments it seems possible that all the slowdown was caused just by `RegexOptions.Compiled` _while_ creating instance per comparison in either case.